framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala

```diff
@@ -0,0 +1,103 @@
+package com.sneaksanddata.arcane.framework
+package models.cdm
+
+import models.{ArcaneSchema, DataCell, DataRow, MergeKeyField}
+
+import scala.language.implicitConversions
+import scala.util.matching.Regex
+
+object CSVParser:
+  def parseCsvLine(line: String, delimiter: Char = ','): Seq[Option[String]] = {
+    def isQuote(position: Int): Boolean = line(position) == '"'
+    def isDelimiter(position: Int): Boolean = line(position) == delimiter
+    def isEol(position: Int): Boolean = position == line.length - 1
+
+    def extractValue(fromIndex: Int, currentPosition: Int): Option[String] =
+      if currentPosition == fromIndex then
+        if isDelimiter(fromIndex) then
+          None
+        else Some(line(currentPosition).toString)
+      else
+        (isDelimiter(currentPosition), isEol(currentPosition)) match
+          // empty value at the end of the line
+          case (true, true) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1)) // if there is a quote at the end, move back by 1 character
+          case (true, true) => Some(line.slice(from = fromIndex, until = currentPosition))
+          case (false, true) if !isQuote(currentPosition) => Some(line.slice(from = fromIndex, until = currentPosition + 1))
+          case (false, true) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1))
+          case (true, false) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1))
+          case _ => Some(line.slice(from = fromIndex, until = currentPosition))
+
+    def scanLine(charIndex: Int, quoteSum: Int, prevCharIndex: Int): IndexedSeq[Option[String]] = {
```

Comment:
Is it supposed to be tail recursion? If yes, annotate it with the `@tailrec` annotations, please.

Reply:
no this function is not tail-recursive

Reply:
Ok. I have a slight concern about the memory consumption by this function, but we can still test it with real data when we run it in the cluster